mis_builder/models/aep.py

```diff
@@ -67,9 +67,11 @@ class AccountingExpressionProcessor(object):
                          r"(?P<accounts>_[a-zA-Z0-9]+|\[.*?\])"
                          r"(?P<domain>\[.*?\])?")
 
-    def __init__(self, company):
-        self.company = company
-        self.dp = company.currency_id.decimal_places
+    def __init__(self, companies, currency_id=None):
+        self.companies = companies
+        self.currency_id = currency_id and currency_id or \
+            self.companies.env.user.company_id.currency_id
```

Comment:
here, raise UserError if currency_id is non and companies don't have all the same currency